JavaScript Capstone

VelzckC0D3 opened the pull request:
# JavaScript Capstone -  `WebPack` `ES6` `API`

## ✨ _Requested Implementations_ ✨
✔️ _Uses Javascript to make a dynamic website with a single page_
✔️ _Uses ES6_
✔️ _Use callbacks and promises_
✔️ _Use webpack_
✔️ _Apply JavaScrip best practices and language style guides in code_
✔️ _Use AAA testing patterns_
✔️ _Unit testing with Jest_
✔️ _Follows GitFlow_
✔️ _Solve simple git conflicts_
✔️ _Send and receive data from an API_
✔️ _Use API documentation_
✔️ _Understand and use JSON_
✔️ _It uses JavaScript asynchronous code_
✔️ _Several code reviews has been made_

#  💡 _Thank you for your review!_ 💡

MarcoDDM committed:
Merge pull request #54 from VelzckC0D3/setup_webpack

Setup webpack

MarcoDDM committed:
Basic structure and styling

MarcoDDM committed:
Change footer content

MarcoDDM committed:
Remove hello webpack mssg

MarcoDDM committed:
Add footer position to bottom

MarcoDDM committed:
Remove > typo

MarcoDDM committed:
Add style to position the footer content

MarcoDDM committed:
Add margin 0 to body

MarcoDDM committed:
Add favicon meta and file

VelzckC0D3 committed:
Merge pull request #55 from VelzckC0D3/creaded_structure

Add styles for page background and footer

VelzckC0D3 committed:
Header

- Implemented the header to the basic HTML and stylized it

VelzckC0D3 committed:
Header

- Checked all linters

VelzckC0D3 committed:
linters

- eslint corrected

MarcoDDM committed:
Changes for fix linters

MarcoDDM committed:
Merge pull request #56 from VelzckC0D3/Header

Header

MarcoDDM committed:
Add function to hit api and css for styling

MarcoDDM committed:
Change for linters

MarcoDDM committed:
Fix linters

MarcoDDM committed:
Fix linters

MarcoDDM committed:
Add comments button

MarcoDDM committed:
Refactor to use async and promises

MarcoDDM committed:
Refactor function to pass linters check

VelzckC0D3 committed:
Merge pull request #57 from VelzckC0D3/created_item-list

Add function to hit api and css for styling

VelzckC0D3 committed:
Pop-Up

- Added dynamic pop up which shows the current information of the game and a comments section
- Prepared the comment form to make the actual function
- Checked all linters
- Personalized

MarcoDDM committed:
Add fetch function to get and submit the likes

MarcoDDM committed:
Add like counter to card

MarcoDDM committed:
Add likes from the back on the card

MarcoDDM committed:
Change on likes message on screen

MarcoDDM committed:
Change to fix linters

MarcoDDM committed:
Add user input on form for comments

MarcoDDM committed:
Add fetch function to get comments and sumbit them

VelzckC0D3 committed:
Dynamic Comments

- Added the function where we're going to work to add the dynamic comments for each game with its respective published date and user

VelzckC0D3 committed:
Fixed Likes Function

- Now the likes are properly updated in real time after clicking the like button

VelzckC0D3 committed:
Add Comments

- Added all the necessary functions to properly add and display the comments for each game on the pop up dynamically

VelzckC0D3 committed:
Bug Correction

- added a function to avoid spamming the comments

VelzckC0D3 committed:
Modules

- added a module with all the JS behavior following ES6 good practices

VelzckC0D3 committed:
Testing & Responsive

- added unit testing for the functions made by @MarcoDDM
- added media queries to make the site responsive
- checked all the linters

VelzckC0D3 committed:
corrected bugs

- fixed a typo and conflict with a folder on github

VelzckC0D3 committed:
Checking Linters

- Trying to fix eslint

VelzckC0D3 committed:
update eslint

- fixed an eslint error

MarcoDDM committed:
Update README.md

MarcoDDM committed:
Update README.md

MarcoDDM committed:
Add mockup file

MarcoDDM committed:
Update README to add mockup

VelzckC0D3 committed:
Update README.md

VelzckC0D3 committed:
Update README.md

MarcoDDM committed:
Add demo loom

VelzckC0D3 committed:
Merge branch 'dev' into Pop-Ups

VelzckC0D3 committed:
Merge pull request #59 from VelzckC0D3/Pop-Ups

Pop ups

AnselemOdims commented on src/module/behaviors.js:
```diff
@@ -0,0 +1,230 @@
+  game.short_description.length > 200 ? '...' : ''
+}</p>
+    <div class="game-actions">
+      <a class="game-link" href="${game.game_url}" target="_blank">Play Now!</a>
```
- [ ] Please kindly remove the `Play now` button as there is no provision for it in the design.

AnselemOdims commented on src/module/behaviors.js:
```diff
@@ -0,0 +1,230 @@
+    renderGameCards(data);
+  });
+};
+
+init();
```
- [ ] Good job working on other functionalities, but please it is important that you implement a counter functionality and include it on every page. The counter functionality should count the number of items in the list on that page.
- [ ] There should be a counter functionality on the homepage that displays the number of items on that page.
- [ ] Also, there should be another one in the comments popup that shows the number of comments for that item.
- [ ] Please also write a unit test to test out the counter functionality.

AnselemOdims reviewed (changes requested):
### Hi @team, 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

MarcoDDM committed:
Remove Play Now btn

MarcoDDM committed:
Remove Comment btn literal

VelzckC0D3 committed:
Counters Functions

- Added a counter function on each page for the games
- Added a counter function on each pop up for the comments

VelzckC0D3 committed:
Merge branch 'dev' of https://github.com/VelzckC0D3/JavaScript-Captsone into dev

VelzckC0D3 committed:
Update README.md
